src/lib/datatypes/sol-buffer.c

```diff
@@ -233,11 +236,11 @@ sol_buffer_append_vprintf(struct sol_buffer *buf, const char *fmt, va_list args)
             goto end;
 
         /* Extra room for the ending NUL-byte. */
-        if (new_size >= SIZE_MAX - nul_byte_size(buf)) {
+        if (new_size >= SIZE_MAX - NUL_SIZE) {
```

Comment:
Minor suggestions: I would change the comment above to say

   /* Extra room for the ending NUL-byte, necessary even when the buffer has
      SOL_BUFFER_FLAGS_NO_NUL_BYTE because vsnprintf always write the '\0'. */

And use the literal 1 or a local constant instead of a define macro.